src/Umbraco.Infrastructure/PropertyEditors/FileUploadPropertyValueEditor.cs

```diff
@@ -108,7 +113,7 @@ internal class FileUploadPropertyValueEditor : DataValueEditor
         }
 
         // process the file
-        var filepath = editorFile == null ? null : ProcessFile(file, editorValue.DataTypeConfiguration, cuid, puid);
+        var filepath = editorFile == null ? null : ProcessFile(file, editorValue.DataTypeConfiguration, _mediaService.GetById(editorValue.ContentKey)!, puid);
```

Comment:
When editorValue.ContentKey is returning an Guid.Empty, this will return null.

And explode later 🙈